File: src/test/scala/pl/aetas/bakspad/presentation/LocaleAwareStringComparatorTest.scala
package pl.aetas.bakspad.presentation

import pl.aetas.bakspad.spec.UnitSpec
import org.junit.runner.RunWith
import org.scalatest.junit.JUnitRunner
import java.util.Locale

@RunWith(classOf[JUnitRunner])
class LocaleAwareStringComparatorTest extends UnitSpec {

  val polishLocaleComparator = new LocaleAwareStringComparator(new Locale("pl_PL"))

  "LocaleAwareStringComparator" should "use locale specific order when returning comparable result" in {
    val first = "ą"
    val second = "z"
    val compare = polishLocaleComparator.compare(first, second)
    compare should be < 0
  }
}

File: src/test/scala/pl/aetas/bakspad/presentation/LocaleAwareStringComparatorTest.java
package pl.aetas.bakspad.presentation;

import org.junit.Before;
import org.junit.Test;

import java.util.Locale;

import static org.hamcrest.Matchers.lessThan;
import static org.junit.Assert.assertThat;

public class LocaleAwareStringComparatorTest {

    private LocaleAwareStringComparator polishLocaleComparator;

    @Before
    public void setUp() throws Exception {
        polishLocaleComparator = new LocaleAwareStringComparator(new Locale("pl_PL"));
    }

    @Test
    public void shouldTakeLocaleSpecificOrderWhenReturningComparableResult() throws Exception {
        String first = "ą";
        String second = "z";

        int compare = polishLocaleComparator.compare(first, second);
        assertThat(compare, lessThan(0));

    }
}
